snaps_k8s/provision/ansible_configuration.py

```diff
@@ -902,3 +913,159 @@ def delete_weave_interface(k8s_conf):
+    """
+    function used to call launch_load_balancer
+    :param hostname_map:
+    :param host_node_type_map:
+    :param loadbalancer_dict:
+    :param project_name:
+    :return:
+    """
+    logger.info("Argument List:\n hostname_map: %s\n host_node_type_map: "
+                "%s\n loadbalancer_dict: %s \n pb_vars: %s",
+                hostname_map, host_node_type_map, loadbalancer_dict,
+                pb_vars)
+
+    logger.info("launch_ha_loadbalancer_conf function")
+    host_node_type_map_local = {}
+    lb_port = str(loadbalancer_dict.get("port"))
+    master_ip_list = []
+    for key in host_node_type_map:
+        if host_node_type_map[key] == "master":
+            master_ip_list.append(hostname_map[key])
+    master_ip_list = str(master_ip_list)
+    logger.info(host_node_type_map_local)
+    logger.info('INSTALL / CONFIGURE EXTERNAL LOAD BALANCER on -- ip %s',
+                host_node_type_map_local.get('localhost'))
+    __launch_load_balancer(
+        consts.K8_HA_EXT_LB,
+        loadbalancer_dict.get(consts.IP_KEY),
+        loadbalancer_dict.get(consts.HOSTNAME_KEY),
+        master_ip_list, lb_port,
+        loadbalancer_ip=loadbalancer_dict.get(consts.IP_KEY),
+        base_pb_vars_ha=pb_vars)
+
+
+def __launch_kubespray_ha_configure(hostname_map=None,
+                                    host_node_type_map=None,
+                                    loadbalancer_dict=None,
+                                    pb_vars=None):
+    """
+    fucntion used to call kubespray_ha_configure
+    :param hostname_map:
+    :param host_node_type_map:
+    :param loadbalancer_dict:
+    """
+    logger.info("Argument List:\n hostname_map: %s\n host_node_type_map: "
+                "%s\n loadbalancer_dict: %s\n ",
+                hostname_map, host_node_type_map, loadbalancer_dict)
+
+    ret_val = False
+    logger.info("launch_ha_kubespray_configure function")
+    host_node_type_map_local = {'localhost': '127.0.0.1'}
+    __kubespray_ha_configure(
+        consts.K8_HA_KUBESPRAY_CONFIGURE,
+        host_node_type_map_local['localhost'], 'localhost',
+        loadbalancer_dict.get(consts.IP_KEY),
+        pb_vars=pb_vars)
+
+
+def __launch_load_balancer(playbook, ip_addr, host_name,
+                           master_ip_list, lb_port,
+                           loadbalancer_ip=None,
+                           base_pb_vars_ha=None):
+    """
+    function added for ha install load balancer
+    :param playbook:
+    :param ip_addr:
+    :param host_name:
+    :param master_ip_list:
+    :param lb_port:
+    :param loadbalancer_ip:
+    """
+    pb_vars = {
+        'ip': ip_addr,
+        'host_name': host_name,
+        'MASTER_IP_LIST': master_ip_list,
+        'lb_port': lb_port,
+        'LOADBALANCER_IP': loadbalancer_ip,
+    }
+    pb_vars.update(base_pb_vars_ha)
+    logger.info("Arguments are %s", pb_vars)
+    ansible_utils.apply_playbook(playbook, [loadbalancer_ip], variables=pb_vars)
```

Comment:
You need to add the user here.
    ansible_utils.apply_playbook(
        playbook, [loadbalancer_ip], consts.NODE_USER, variables=pb_vars)

Reply:
fixed in 'ha_ci_scripts' branch